eclipse/src/saros/ui/wizards/pages/LocalRepresentationSelectionPage.java

```diff
@@ -0,0 +1,703 @@
+   *
+   * <p>If the reference point selection tab selection is valid, the error message entry is dropped
+   * (if present). Otherwise, the new error message is entered into the list of current errors. Any
+   * existing error messages for the tab are overwritten.
+   *
+   * @param referencePointID the ID of the reference point tab to update
+   * @see #isReferencePointRepresentationSelectionValid(String)
+   */
+  private void updateReferencePointSelectionStatus(String referencePointID) {
+    String errorMessage = isReferencePointRepresentationSelectionValid(referencePointID);
+
+    if (errorMessage != null) {
+      currentErrors.put(referencePointID, errorMessage);
+
+    } else {
+      currentErrors.remove(referencePointID);
+    }
+  }
+
+  /**
+   * Scans the current Eclipse Workspace for existing project artifacts that would clash with the
+   * projects created as part of the resource negotiation with the current selections.
+   *
+   * @return a warning message if such artifacts are found, or <code>null</code> otherwise
+   */
+  private String findAndReportClashingProjectArtifacts() {
+    IPath workspacePath = ResourcesPlugin.getWorkspace().getRoot().getLocation();
+
+    if (workspacePath == null) {
+      return null;
+    }
+
+    File workspaceDirectory = workspacePath.toFile();
+
+    List<String> dirtyProjectNames = new ArrayList<>();
+
+    for (ReferencePointOptionComposite composite : referencePointOptionComposites.values()) {
+      ReferencePointOptionResult referencePointOptionResult = composite.getResult();
+
+      if (referencePointOptionResult.getLocalRepresentationOption() != NEW_PROJECT) {
+        continue;
+      }
+
+      String projectName = referencePointOptionResult.getNewProjectName();
+
+      if (projectName == null || projectName.isEmpty()) {
+        continue;
+      }
+
+      if (new File(workspaceDirectory, projectName).exists()) {
+        dirtyProjectNames.add(projectName);
+      }
+    }
+
+    String warningMessage = null;
+
+    if (!dirtyProjectNames.isEmpty()) {
+      warningMessage =
+          MessageFormat.format(
+              Messages.LocalRepresentationSelectionPage_warning_project_artifacts_found,
+              StringUtils.join(dirtyProjectNames, ", "));
+    }
+
+    return warningMessage;
+  }
+
+  /**
+   * Sets preselected values for all contained reference point option composites and then runs the
+   * input validation. The preselected values do not have to be valid.
+   *
+   * <p>Sets the reference point name as the default value for the new project and new directory
+   * name.
+   *
+   * <p>If there is a previous path mapping for the reference point name, the option to use an
+   * existing directory using that path is selected. Otherwise, if the reference point has the same
+   * name as a project in the local workspace, the option to use an existing directory using the
+   * project path is selected.
+   */
+  private void preselectValues() {
```

Comment:
I guess the logic is the same as the old one. So my question is if it is still necessary to pre-select an existing directory and afterwards disable the option to change it.

I do not see any issue to map  /A/foo to /B/foo and /A/bar to /C/bar instead of /B/bar or is it possible to share /A after you shared /A/foo and /A/bar ?

Reply:
This logic only forces the selection of already shared reference points, i.e. reference points that were already shared before the negotiation.

I don't know why this option was added. I don't think the case it is covering should be possible. But I did not see a reason to remove it in case it covers some weird corner case.

Reply:
It was added because of partial sharing. E.g you start only with 1 files. Afterwards you add more files. The core logic was not capable of mapping new files that belongs to a shared project of the remote side to a complete different project because every resource was tied to the project id and therefore it was a nobrainer to preselect the project and disallow any changes at all.

But as you already mentioned this should not happen at all any longer, i.e the logic will never triggered.

Reply:
Ah, ok. Thanks for the explanation. Will remove it.

Reply:
Done.